DatadogCore/Sources/Core/Context/DatadogContextProvider.swift

```diff
@@ -110,7 +108,12 @@ internal final class DatadogContextProvider {
     ///
     /// - Parameter block: The block closure called with the current context.
     func write(block: @escaping (inout DatadogContext) -> Void) {
-        queue.async { block(&self.context) }
+        queue.async {
```

Comment:
Given what you mentioned in the PR description, should we check if the context changed before notifying the receivers?

Reply:
this is not so possible with current context definition, we are going to do some followup with a larger refactor.